plugin/pkg/shoot/validator/admission.go

```diff
@@ -766,6 +768,78 @@ func validateMachineImagesConstraints(constraints []core.MachineImage, image, ol
 	return false, validValues
 }
 
+func validateContainerRuntimeConstraints(constraints []core.MachineImage, worker, oldWorker core.Worker, fldPath *field.Path) field.ErrorList {
+	if apiequality.Semantic.DeepEqual(worker.CRI, oldWorker.CRI) {
+		return nil
+	}
+
+	if worker.CRI == nil {
+		return nil
+	}
+
+	var machineImage *core.MachineImage
+	var machineVersion *core.MachineImageVersion
+
+	for _, image := range constraints {
+		if image.Name == worker.Machine.Image.Name {
```

Comment:
Hm, nil check for `image`?

Reply:
It is highly unlikely to happen due to the defaulting, but I have added a check just in case.